auslib/test/test_db.py

```diff
@@ -3919,11 +3915,110 @@ def testAllTablesExist(self):
+        self.aus_db_tables = [self.db.rulesTable, self.db.releasesTable,
+                              self.db.permissionsTable, self.db.dockerflowTable]
+
+    def _is_column_unique(self, col_obj):
+        """
+        Check to see if a column is unique using a Sqlite-specific query.
+        """
+        col_engine = col_obj.table.metadata.bind
+        table_name = col_obj.table.name
+        res = col_engine.execute("SELECT sql FROM sqlite_master WHERE name = :table_name", table_name=table_name)
+        res = res.fetchone()[0]
+
+        if re.search(r'(?:CONSTRAINT (\w+) +)?UNIQUE *\({0}\)'.format(col_obj.name), res) is None:
+            return None
+        return True
+
+    def _get_migrated_db(self):
+        db = AUSDatabase('sqlite:///' + self.getTempfile())
+        db.create()
+        return db
+
+    def _get_reflected_metadata(self, db):
+        """
+        @type db: AUSDatabase
+        """
+        mt = MetaData()
+        engine = create_engine(db.dburi)
+        mt.bind = engine
+        mt.reflect()
+        return mt
+
+    def assert_attributes_for_tables(self, tables):
+        """
+        Expects an iterable of sqlalchemy (Table, Table) pairs. The first table
+        should be the result of migrations, the second should be the model
+        taken directly from db.py.
+        """
+        failures = []
+        for reflected_table, table_model_instance in tables:
+
+            for col_name in table_model_instance.c.keys():
+                db_py_col = table_model_instance.c[col_name]
+                reflected_db_col = reflected_table.c[col_name]
+
+                for col_property in self.properties:
+                    db_py_col_property = getattr(db_py_col, col_property)
+                    reflected_db_col_property = getattr(reflected_db_col, col_property)
+
+                    if db_py_col_property != reflected_db_col_property:
+                        failures.append(
+                            self.property_err_msg.format(
+                                property=col_property,
+                                table_name=table_model_instance.name,
+                                column=col_name,
+                                model_prop=db_py_col_property,
+                                reflected_prop=reflected_db_col_property))
+
+                # Testing 'unique' separately since Sqlalchemy < 1.0.0 can't reflect this attribute for this version of sqlite
+                ref_uniq = self._is_column_unique(reflected_db_col)
+                if db_py_col.unique != ref_uniq:
+                    failures.append(
+                        self.property_err_msg.format(
+                            property="unique",
+                            table_name=table_model_instance.name,
+                            column=col_name,
+                            model_prop=db_py_col.unique,
+                            reflected_prop=ref_uniq))
+
+        self.assertEqual(failures, [], str(failures))
```

Comment:
I like that we have all of the errors in one assert, but I wonder if there's a way to improve the formatting such that it includes newlines - it's tough to read right now. Not a blocker, just nice to have if it's easy.